### PR TITLE
Split up healthcheck and liveness probes

### PR DESCRIPTION
**Pull request by halkeye**
healthcheck = makes sure app is still running
liveness = makes sure credentials are all right and such

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -35,30 +36,42 @@ app.use(bodyParser.urlencoded({ extended: false }))
...
-  if (config.JENKINS_AUTH) {
+const healthchecks = {
+  jenkins: async function () {
+    const jenkinsOpts = {};
+    if (!config.JENKINS_AUTH) {
+      return { jenkins: 'no_auth' };
+    }
+
     jenkinsOpts.headers = {'Authorization': 'Basic ' + new Buffer.from(config.JENKINS_AUTH, 'utf8').toString('base64')};
+    const response = await fetch(config.JENKINS_HOST + '/whoAmI/api/json', jenkinsOpts)
```
Comment (timja) [inline index.js:47]: how often is this triggered? and will restarting it help at all? (liveness)

Comment (halkeye) [reply index.js:47]: I mean its configurable, but ~liveness~readyness is for when k8s makes sure the pod is ready. I figure thats the only time it makes sense to do credentials checking.

I guess i should actually rename them, i was intending this to be the startup check

File: index.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ const healthchecks = {
   }
 }
 
-app.use('/liveness', asyncWrap(async (req, res) => {
+app.use('/readyness', asyncWrap(async (req, res) => {
```
Comment (timja) [inline index.js:58]:
```suggestion
app.use('/readiness', asyncWrap(async (req, res) => {
```

**Review by timja — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (timja) [reply index.js:49]: startup check isn't available on our version of k8s, readiness check is still called all the time.

File: index.js (hunk 2 above)
Comment (halkeye) [reply index.js:58]: Rawr can I not accept on the phone

File: index.js (hunk 1 above)
Comment (halkeye) [reply index.js:49]: Really? That seems dumb. I know there are two probes. I've never tried setting them to different things before

Ite probably a good thing to have both and have Prometheus or something else  check the expensive one

Comment (timja) [reply index.js:49]: startup probes are available from 1.16 which should be okay to upgrade to but might be better to wait for olbak 👅 .

and yeah we've been bitten by assuming readiness was only called on startup.

**Pull request merged**